ledger/simulation/simulation_eval_test.go

```diff
@@ -1170,11 +1170,135 @@ func TestInvalidTxGroup(t *testing.T) {
 	}
 
 	// should error with invalid transaction group error
-	_, err := s.Simulate(txgroup)
+	_, err := s.Simulate(simulation.Request{TxnGroups: [][]transactions.SignedTxn{txgroup}})
 	require.ErrorAs(t, err, &simulation.InvalidTxGroupError{})
 	require.ErrorContains(t, err, "transaction from incentive pool is invalid")
 }
 
+// TestLogLimitLiftingInSimulation tests that an app with log calls that exceed limits during normal runtime
+// can get through during simulation with `lift-log-limits` activated
+func TestLogLimitLiftingInSimulation(t *testing.T) {
```

Comment:
Testing comments:
* You can use `simulationTest` to simplify setup and comparison of expected vs actual result. Not all tests in this file use it because some only want to verify part of the response, but I think generally speaking its stronger to compare the entire response if not too much work
* It would be nice to see at least one `log` with a value longer than 1024, which is the normal maximum (the `bzero` opcode might be helpful if you don't care about the byte contents)
* I'd also like to see a test of what happens if you surpass even the higher logging limits

Reply:
see [updated eval test, repeat 40 times line 1050 length, eval-changes to …](https://github.com/algorand/go-algorand/pull/5247/commits/04436a3ebb3923b05ccc7a0614f7d4bc7254e057), I augmented the line to length 1050 and repeated 40 times in this test.

Reply:
The reason I am not using `simulationTest` in this case is that, it is only for vanilla case, with no other power packs.

I will probably at some point augment the definition to allow for passing in the simulation related config for powerpacks.

Reply:
Good point, we should change `simulationTest` in the future to accommodate the power packs, but it doesn't have to happen right now.

Still interested in seeing a test which exceeds even the higher log limits, that's the only thing holding be back from approving

Reply:
I see, so you are trying to hit 65536 in bytes or 2048 in calls as counter cases? Will do in next commit.

Reply:
Whichever is easier

Reply:
see [exceeding 65536 byte length](https://github.com/algorand/go-algorand/pull/5247/commits/5b4c05f4dd9ca2a91200d986b13c7b91d7614f77)